### PR TITLE
Core/GameObject: some GO fixes (Rituals, LoS, Traps and size calculation)

### PR DESCRIPTION
**Pull request by ghost**
**Target branch(es):** 3.3.5

- [x] 3.3.5

**Tests performed:** Compiles, simple ingame tests

Fixed rituals and properly implemented animation usage and checking if channel is maintained
Properly link traps for Button Gameobjects
Moved execution of chests event scripts to loot release, fixes possible exploits and works blizzlike now
Changed collision toggle conditions for gameobjects, they work as intended now
Take into account actual object scale when calculating size

To do:
- [ ] animation spell ids in sql file for warlock rituals and summoning stones may be improper - sniff proper ids

### CODE REVIEW

File: sql/updates/world/3.3.5/9999_99_99_99_world.sql [hunk 1]
```diff
@@ -0,0 +1,4 @@
+
+UPDATE `gameobject_template` SET `data2`=32783, `data3`=0 WHERE `entry` IN(36727, 179944, 194108); -- Summoning stones and Ritual of Summoning animation spell
```
Comment (Aokromes) [inline sql/updates/world/3.3.5/9999_99_99_99_world.sql]: change data fields without sniff is hack.

Comment (xinef1) [inline sql/updates/world/3.3.5/9999_99_99_99_world.sql]: Mentioned in to do section

**Comment by jackpoz** [comment]
what should we do with this PR ?

**Comment by tkrokli** [comment]
We usually put PRs like this on hold (close it for now) and ask openly for someone else to reuse the code in a new PR if there is a chance that the new PR will be maintained and updated with documented or approved content.

**Comment by Killyana** [comment]
Must fix https://github.com/TrinityCore/TrinityCore/issues/15453

**Comment by ariel-** [comment]
What about merging it without the WDB modifications? or research them

**Comment by Killyana** [comment]
I checked many sniffs and this spells doesn't appears, but there's always something related to the channeling, without any spell:
```
ServerToClient: SMSG_COMPRESSED_UPDATE_OBJECT (0x01F6) Length: 56 ConnIdx: 0 Time: 07/15/2010 21:13:54.000 Number: 182547
Count: 3
[0] UpdateType: Values
[0] GUID: Full: 0xXXXXXX1 Type: Player Low: X1Name: X1
[0] UNIT_FIELD_CHANNEL_OBJECT: 0/0
[0] 21: 0/0
[0] UNIT_CHANNEL_SPELL: 0/0
[1] UpdateType: Values
[1] GUID: Full: 0xXXXXXX2 Type: Player Low: 9257970 Name: X2
[1] UNIT_FIELD_CHANNEL_OBJECT: 0/0
[1] 21: 0/0
[1] UNIT_CHANNEL_SPELL: 0/0
[2] UpdateType: Values
[2] GUID: Full: 0xXXXXX3 Type: Player Low: 34972670 Name: X3
[2] UNIT_FIELD_CHANNEL_OBJECT: 0/0
[2] 21: 0/0
[2] UNIT_CHANNEL_SPELL: 0/0
[2] UNIT_FIELD_POWER1: 20823/2.917924E-41

**Comment by jackpoz** [comment]
Closing the PR, it can be easily found with the "Closed-*" labels. If anyone would like to finish it they can ask for it to be reopened.

**Pull request closed without merging**